realm/realm-library/src/main/cpp/io_realm_SyncSession.cpp

```diff
@@ -161,20 +161,19 @@ JNIEXPORT jboolean JNICALL Java_io_realm_SyncSession_nativeWaitForDownloadComple
                                                         "(ILjava/lang/Long;Ljava/lang/String;)V");
             JavaGlobalRef java_session_object_ref(env, session_object);
 
-            bool listener_registered =
-                session->wait_for_download_completion([java_session_object_ref, callback_id](std::error_code error) {
-                    JNIEnv* env = JniUtils::get_env(true);
-                    JavaLocalRef<jobject> java_error_code;
-                    JavaLocalRef<jstring> java_error_message;
-                    if (error != std::error_code{}) {
-                        java_error_code =
-                            JavaLocalRef<jobject>(env, JavaClassGlobalDef::new_long(env, error.value()));
-                        java_error_message = JavaLocalRef<jstring>(env, env->NewStringUTF(error.message().c_str()));
-                    }
-                    env->CallVoidMethod(java_session_object_ref.get(), java_notify_result_method,
-                                        callback_id, java_error_code.get(), java_error_message.get());
-                });
-
+            // FIXME
+            std::function<void(std::error_code)> callback = [&java_session_object_ref, callback_id](std::error_code error) {
```

Comment:
`&java_session_object_ref` capture the ref here seems to be wrong. 
the `callback` could be called after `Java_io_realm_SyncSession_nativeWaitForDownloadCompletion` returns. If that happens, `java_session_object_ref` will be out of scope. I think the original code is correct.

Reply:
I changed it because Clang didn't like the original code:
```
/usr/local/opt/android-ndk-r18b/sources/cxx-stl/llvm-libc++/include/memory:2139:9: error: no matching constructor for initialization of '(lambda at /Users/cm/Realm/realm-java/realm/realm-library/src/main/cpp/io_realm_SyncSession.cpp:165:55)'
      : __value_(_VSTD::forward<_Args>(_VSTD::get<_Indexes>(__args))...) {}
```

Guess I need to take a 2nd look